wdl/downsampling_pipeline_part2.wdl

```diff
@@ -1,534 +1,638 @@
+      fastq_file_2 = fastq_file_2,
+      downsample_docker = downsample_docker,
+      start_depth = start_depth,
+      final_depth = final_depth,
+      seed = select_first([seed_override, 20937]),
+      runtime_attr_override = runtime_attr_random_sample
+    }
+
+
+  #################################################################################
+  ####        bwa realigns the downsampled fastq files to Hg38 genome,            #
+  ####        adds read groups and mark duplicates and converts to cram           #
+  #################################################################################
+  call realign {
+    input :
+      downsample_file_1 = countAndRandomSample.downsample_file_1,
+      downsample_file_2 = countAndRandomSample.downsample_file_2,
+      reference_fasta = reference_fasta,
+      downsample_docker = downsample_docker,
+      runtime_attr_override = runtime_attr_realign,
+      final_depth = final_depth,
+      ref_amb = ref_amb,
+      ref_ann = ref_ann,
+      ref_bwt = ref_bwt,
+      ref_pac = ref_pac,
+      ref_sa = ref_sa,
+      ref_fai = ref_fai,
+      ref_dict = ref_dict
+  }
+
+  #################################################################################
+  ####        Re-adds Read Groups back to realigned bam files                     #
+  #################################################################################
+  call addReadGroupAndSort {
+    input :
+      bam_downsample_file = realign.bam_downsample_file,
+      downsample_docker = downsample_docker,
+      original_cram_or_bam_file_read_groups = original_cram_or_bam_file_read_groups,
+      runtime_attr_override = runtime_attr_add_read_group
+  }
+
+  #################################################################################
+  ####        Marks duplicates to closely mimick format of sequencers             #
+  #################################################################################
+  call markDuplicatesAndToCram {
+    input :
+      bam_sorted_rg_file = addReadGroupAndSort.bam_sorted_rg_file,
+      downsample_docker = downsample_docker,
+      runtime_attr_override = runtime_attr_mark_duplicates,
+      reference_fasta = reference_fasta
+  }
+
+
+  #################################################################################
+  ####        Sorts and indexes realigned sequences and convert to crams          #
+  #################################################################################
+  call sortIndex {
+    input :
+      cram_downsample_file = markDuplicatesAndToCram.cram_downsample_file,
+      downsample_docker = downsample_docker,
+      runtime_attr_override = runtime_attr_sort_index
+  }
+
+
+  #################################################################################
+  ####        Checks for downsamples sequences via picard                         #
+  #################################################################################
+  call countCoverage {
```

Comment:
To follow up on a thread from my first review - you explained that "collectCoverage is used to double check that the downsampled file has the final read depth that we want for each run". In that case, it sounds like it's more for QC purposes while you're developing the pipeline? If it won't necessarily be needed for every sample in the scaled-up pipeline, you could consider adding a boolean input `run_count_coverage` to toggle it on/off as needed. If it's something you want to run every time, then ignore this.